src/java.desktop/macosx/classes/sun/lwawt/macosx/CClipboard.java

```diff
@@ -89,7 +89,11 @@ protected void setContentsNative(Transferable contents) {
 
             try {
                 byte[] bytes = DataTransferer.getInstance().translateTransferable(contents, flavor, format);
-                setData(bytes, format);
+                if (DataFlavor.javaFileListFlavor.equals(flavor)) {
+                    writeFileObjects(bytes);
```

Comment:
looks to me like you are supposed to pass the format too, the native code seems to expect it.